sky/tools/create_full_ios_framework.py

```diff
@@ -0,0 +1,214 @@
+        os.path.join(buildroot_dir, args.armv7_out_dir)
+    )
+
+  simulator_x64_out_dir = None
+  if args.simulator_x64_out_dir:
+    simulator_x64_out_dir = (
+        args.simulator_x64_out_dir if os.path.isabs(args.simulator_x64_out_dir)
+        else os.path.join(buildroot_dir, args.simulator_x64_out_dir)
+    )
+
+  framework = os.path.join(dst, 'Flutter.framework')
+  simulator_framework = os.path.join(dst, 'sim', 'Flutter.framework')
+  arm64_framework = os.path.join(arm64_out_dir, 'Flutter.framework')
+  simulator_x64_framework = os.path.join(
+      simulator_x64_out_dir, 'Flutter.framework'
+  )
+  if args.simulator_arm64_out_dir is not None:
+    simulator_arm64_framework = os.path.join(
+        args.simulator_arm64_out_dir, 'Flutter.framework'
+    )
+    simulator_arm64_dylib = os.path.join(simulator_arm64_framework, 'Flutter')
+
+  arm64_dylib = os.path.join(arm64_framework, 'Flutter')
+  simulator_x64_dylib = os.path.join(simulator_x64_framework, 'Flutter')
+
+  if not os.path.isdir(arm64_framework):
+    print('Cannot find iOS arm64 Framework at %s' % arm64_framework)
+    return 1
+
+  if not os.path.isdir(simulator_x64_framework):
+    print('Cannot find iOS x64 simulator Framework at %s' % simulator_framework)
+    return 1
+
+  if not os.path.isfile(arm64_dylib):
+    print('Cannot find iOS arm64 dylib at %s' % arm64_dylib)
+    return 1
+
+  if not os.path.isfile(simulator_x64_dylib):
+    print('Cannot find iOS simulator dylib at %s' % simulator_dylib)
+    return 1
+
+  if not os.path.isfile(DSYMUTIL):
+    print('Cannot find dsymutil at %s' % DSYMUTIL)
+    return 1
+
+  shutil.rmtree(framework, True)
+  shutil.copytree(arm64_framework, framework)
+  framework_binary = os.path.join(framework, 'Flutter')
+  process_framework(args, framework, framework_binary)
+
+  if args.simulator_arm64_out_dir is not None:
+    shutil.rmtree(simulator_framework, True)
+    shutil.copytree(simulator_arm64_framework, simulator_framework)
+
+    simulator_framework_binary = os.path.join(simulator_framework, 'Flutter')
+
+    # Create the arm64/x64 simulator fat framework.
+    subprocess.check_call([
+        'lipo', simulator_x64_dylib, simulator_arm64_dylib, '-create',
+        '-output', simulator_framework_binary
+    ])
+    process_framework(args, simulator_framework, simulator_framework_binary)
+    simulator_framework = simulator_framework
+  else:
+    simulator_framework = simulator_x64_framework
+
+  # Create XCFramework from the arm-only fat framework and the arm64/x64 simulator frameworks, or just the
+  # x64 simulator framework if only that one exists.
+  xcframeworks = [simulator_framework, framework]
+  create_xcframework(location=dst, name='Flutter', frameworks=xcframeworks)
+
+  # Add the x64 simulator into the fat framework
+  subprocess.check_call([
+      'lipo', arm64_dylib, simulator_x64_dylib, '-create', '-output',
+      framework_binary
+  ])
+
+  process_framework(args, framework, framework_binary)
```

Comment:
Can you pull this out into a function called `create_framework` or something similar so it's the same abstraction level as `generate_gen_snapshot` and `zip_archive`?